File: app/src/main/java/ua/com/programmer/simpleremote/specialItems/DocumentField.java
package ua.com.programmer.simpleremote.specialItems;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.Objects;

import ua.com.programmer.simpleremote.Utils;

public class DocumentField {

    private String meta;
    public String type;
    public String name;
    public String description;
    public String code;
    public String value;

    private final Utils utils = new Utils();

    public DocumentField(String init){
        setDefaults();

        if (init.equals("")) return;

        try {
            JSONObject jsonObject = new JSONObject(init);
            initializeFromJSON(jsonObject);
        }catch (JSONException ex){
            utils.log("e", "DocumentField init from string: "+ex.toString());
        }
    }

    public DocumentField(JSONObject jsonObject){
        setDefaults();
        initializeFromJSON(jsonObject);
    }

    private void setDefaults(){
        meta = "unknown";
        type = "";
        name = "";
        description = "";
        code = "";
        value = "";
    }

    private void initializeFromJSON(JSONObject jsonObject){
        JSONArray columnNames = jsonObject.names();
        for (int i = 0; i < Objects.requireNonNull(columnNames).length(); i++) {
            try {
                String columnName = columnNames.getString(i);
                String elementValue = jsonObject.getString(columnName);

                if (columnName.equals("meta")) meta = elementValue;
                if (columnName.equals("name")) name = elementValue;
                if (columnName.equals("type")) type = elementValue;
                if (columnName.equals("description")) description = elementValue;
                if (columnName.equals("code")) code = elementValue;
                if (columnName.equals("value")) value = elementValue;

            }catch (JSONException ex){
                utils.log("e", "DocumentField init from JSON: "+ex.toString());
            }
        }
    }

    public String getNamedValue(){
        return description+": "+value;
    }

    public boolean hasValue(){
        return !value.equals("");
    }

    public boolean isReal() { return !name.equals(""); }

    public String asString(){
        JSONObject jsonObject = new JSONObject();
        try{
            jsonObject.put("meta",meta);
            jsonObject.put("type",type);
            jsonObject.put("name",name);
            jsonObject.put("description",description);
            jsonObject.put("code",code);
            jsonObject.put("value",value);
        }catch (JSONException ex){
            utils.log("e","DocumentField:asString: "+ex.toString());
        }
        return jsonObject.toString();
    }

    public boolean isCatalog(){
        return meta.equals("reference");
    }

    public boolean isDate(){
        return meta.equals("date");
    }
}

File: app/src/main/java/ua/com/programmer/simpleremote/deprecated/specialItems/DocumentField.kt
package ua.com.programmer.simpleremote.deprecated.specialItems

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import ua.com.programmer.simpleremote.deprecated.utility.Utils
import java.util.Objects

class DocumentField {
    private var meta: String? = null
    var type: String? = null
    var name: String? = null
    var description: String? = null
    var code: String? = null
    var value: String? = null

    private val utils = Utils()

    constructor(init: String) {
        setDefaults()

        if (init == "") return

        try {
            val jsonObject = JSONObject(init)
            initializeFromJSON(jsonObject)
        } catch (ex: JSONException) {
            utils.log("e", "DocumentField init from string: $ex")
        }
    }

    constructor(jsonObject: JSONObject) {
        setDefaults()
        initializeFromJSON(jsonObject)
    }

    private fun setDefaults() {
        meta = "unknown"
        type = ""
        name = ""
        description = ""
        code = ""
        value = ""
    }

    private fun initializeFromJSON(jsonObject: JSONObject) {
        val columnNames = jsonObject.names()
        for (i in 0 until Objects.requireNonNull<JSONArray?>(columnNames).length()) {
            try {
                val columnName = columnNames!!.getString(i)
                val elementValue = jsonObject.getString(columnName)

                if (columnName == "meta") meta = elementValue
                if (columnName == "name") name = elementValue
                if (columnName == "type") type = elementValue
                if (columnName == "description") description = elementValue
                if (columnName == "code") code = elementValue
                if (columnName == "value") value = elementValue
            } catch (ex: JSONException) {
                utils.log("e", "DocumentField init from JSON: $ex")
            }
        }
    }

    fun getNamedValue(): String {
        return "$description: $value"
    }

    fun hasValue(): Boolean {
        return value != ""
    }

    fun isReal(): Boolean {
        return name != ""
    }

    fun asString(): String {
        val jsonObject = JSONObject()
        try {
            jsonObject.put("meta", meta)
            jsonObject.put("type", type)
            jsonObject.put("name", name)
            jsonObject.put("description", description)
            jsonObject.put("code", code)
            jsonObject.put("value", value)
        } catch (ex: JSONException) {
            utils.log("e", "DocumentField:asString: $ex")
        }
        return jsonObject.toString()
    }

    fun isCatalog(): Boolean {
        return meta == "reference"
    }

    fun isDate(): Boolean {
        return meta == "date"
    }
}
